### DISCUSSION TITLE
Cubase 11 drops/crackles on large project, unable to use ASIO

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#604] [Help]
Hi,

So on large projects using CUBASE 11 (Mac Mojave) I experience some drops/crackles, and activating ASIO makes things works.

### About drops/crackles :

I was able to trace some of the drops/crackles on specific core spikes (of course you can have 20% cpu usage but one core at 100%). Some metering in the monitor will be really helpfull to detect that :)

Sometimes it is not related to that :( Rebooting everything make things works for a few minutes perfectly then it start dropping again.

**### About ASIO** 

Activating ASIO in a project create drops/crackles, whereas the same project runs fins without ASIO.
Problem is that without ASIO it is almost not possible to use any local plugins :(
It seems that without ASIO I see the CPU/IO gauge increasing a bit sometimes but without reaching max or creating a drop.
With ASIO enabled it runs fluently then I got audio drops after the gauge increase.

Servers are running windows 10 (real time antivirus protection disabled because it can create drops sometimes ...).

Everything running on multiple servers using gigabit ethernet and 10Gbit large switch, network is running perfectly.

**Comment by lesbrunn** [comment]
I'm interested in the solution to this, because I also get dropouts.

**Comment by Henri75** [comment]
By the way there is à second issue (may be related) with cubase : when you
bypass an AG  plugin métiers becope crazy (lot of peaks) with a really
higher ASIO Load and you have drops/crackles.

If i disabled the AG insert instead of bypass everything is fine. If i  en
able again the insert it is working. Problems us with bypass (which makes
AG really difficult to use as you can't bypass any AG plugin without having
the whole projet dropping or crackling

**Reply by TorryCourte** [reply]
Yes, it seems like in bypass, it throws the processing load back to within Cubase. On Big Sur, I also get random security prompts for AudioGridder, even though it's got full Accessibility access rights in the security settings.

**Comment by Henri75** [comment]
Hi Guys,
Any news about those 2 issues (crackling/droping and loop disconnection ?)
AG is amazing but for now quite impossible to really use because of that.
It is really frustrating because it is really great when it works !!!!

**Comment by Henri75** [comment]
I dont think it is à Load in cubase as plugins run remotely.
Mostly a wrong feedback from the plugin when it is bypassed that makes
cubase wait for it ?

**Comment by apohl79** [comment]
Hey, sorry for not responding earlier... Have you ever looked at the plugin or server log files when these dropout happen? If not, could you provide some? or check if you see any buffer warnings or other errors? Is this behaviour the same with small and large buffer sizes in AG?

> when you bypass an AG  plugin métiers becope crazy (lot of peaks) with a really
higher ASIO Load and you have drops/crackles.

Do you mean you are bypassing an inserted remote plugin WITHIN AG or bypassing the AG instance in cubase?

**Comment by Henri75** [comment]
Hi,
Clearly it happens whatever the buffer size is.

I dont have logs anymore as i dont use it for now because of those bug...
Which is à pity because AG is amazing and when it works it is a game
changer !

About the bypass it is when you bypass thé AG plugin itself in Cubase. Then
the Load in cubase increase x times and you got drop. When you reactivate
AG plugin everything is good again and the Load drops drastically !

**Comment by Henri75** [comment]
Same things happens with Asio on or off.
What is weird is that the Load become bigger but you have less/shprter drop
without Asio with Asio you have à little lower Load but a lot of fréquent
and long drop.

It seems that the DAW is getting some wrong or inconsistant data from AG
plugin that it cant handle. I never had such problem with any other DAW

If you need help to test some build or reproduce the problem i can help :)

**Reply by apohl79** [reply]
just a clarification for me, when you say enable or disable asio, you mean the asio-guard under "studio setup"/"vst audio system"? or do you mean the asio-guard plugin setting?

it is weird indeed! could you do the following (if possible I'd like to look at some logs, as they have some statistics):

1) disable tracing in the plugin and in the server settings
2) stop server and daw & wipe all log directories (see here https://audiogridder.com/bug-report-diagnostics-locations/)
3) start server and cubase, disable asio and load your session, hit play and let it play for 10+ seconds (or longer until you hear drops)
4) shutdown everything, package the logs

now repeat 1-4 but in (3) ENABLE asio.

I hope that this provides some info.

Can you reproduce this behaviour with a small project in cubase? Or does it happen on larger projects only? If you have the problem on small projects, could you share some more details about how I can setup a similar project?

I'm still on cubase 10 but already on big sur. It runs very stable for me on cubase.  and btw, I have never deactivated the asio-guard in my studio-setup or the AG plugin.

**Reply by apohl79** [reply]
10s is a bit short. lets say 1min.

**Reply by apohl79** [reply]
Testing with cubase and asio guard a bit over here. 
- My test project has 27 tracks / AG plugins loaded. 
- The server has sandboxing enabled.
- I have a total of 60 remote plugins loaded on AG server (running on a 2018 macmini). 
- I have one plugin UI constantly open. 
- The server load is constantly at 37% with the plugin open and capturing the screen.
- AG buffers with 4 blocks -> +43ms (256 samples I/O buffer on my audio device). 
- It is constantly streaming audio with ~18 MB/s to the server and receiving ~20 MB/s (a bit more because of the plugin UI)
- With asio-guard cubase is showing an audio processing load of ~10-15%, real time peak 0-1%
- Without asio-guard the audio processing load reduces to ~5-10%, real time peak goes up to ~10%
- No audio drops at any point

**Comment by Henri75** [comment]
Please see hereunder

On Fri, 3 Sep 2021, 20:17 Andreas Pohl, ***@***.***> wrote:

> just a clarification for me, when you say enable or disable asio, you mean
> the asio-guard under "studio setup"/"vst audio system"? or do you mean the
> asio-guard plugin setting?
>

There is two different problem :
-  AG plugin bypass when insert in DAW : when it is bypass, i got drops and
peak in Load. When i enable it everything is fine.
- when i use several instance of AG inserted as plugin in Cubase, when i
activate Asio Guard (any level) i got very quickly excessive Load and lot
of audio drops.
When i desactivate, everything is fine.
It looks as if when i activate Asio Guard i have far worst performance
(which should be the opposite),  bigger Asio load in the performance gauge,
and lot of drops. I can use à lot more AG inserts in Cubase without Asio
than with Asio.
Moreover the excessive Asio Load when asio Guard is enable seems fake,
because it is higher than without Asio Guard (again, it should be the
opposite).

Currently i have only large projets (100-200+ tracks) with 10-40 AG
inserts.

Projects runs far better without Asio guard for the AG inserts only, which
is à problem because without Asio Guard you cant really use local plugins
without excessive Load.

I will make a test for you tommorrow.

Any way, the crazy drops and excessive Load in cubase happens even on small
Projects as soon as i bypassan  AG inserts In cubase !
So it is différent but may be related. When i re nable the AG insert
everything is fine again.

**Reply by apohl79** [reply]
Ok will try to reproduce the bypass problem.

**Reply by apohl79** [reply]
And if you get a chance to create these logs (see my message earlier), that might be helpful as well, thx

**Reply by apohl79** [reply]
I short video might be interesting too

**Reply by apohl79** [reply]
> Ok will try to reproduce the bypass problem.

really nothing happens here when bypassing an AG plugin, no hickups/dropouts.. no changes in the performance meter. It just continues to play.

**Reply by apohl79** [reply]
btw, are you using 64bit precision? Thats where I see issues here.

edit: with 64bit precision I get issues/dropouts with large buffers sizes in AG. the audio performance load increases when I bypass AG plugins. and the higher the buffer in AG, the worse it gets. 

So if you are using 64bit, can you try smaller buffers or even try 32bit precision?

**Comment by Henri75** [comment]
Yes i am in 64 bits

**Reply by apohl79** [reply]
The bypass issue will be fixed in the next release. The other issue you have, is not reproducible on my end. So as i wrote, if you can provide logs, i might get more insight.

**Comment by Henri75** [comment]
Great for the bypass fix !

I didnt had time today to do the stuff for the log, i will do my best for
tommorow i promise

**Comment by Henri75** [comment]
Hi,

So great news, I opened some old projects with AG beta 3 and with beta 4...
it didnt happen anymore :)))))

So it was already fixed ! ASIO is working as it should with AG in beta 4 !

I am impatient to have your update with the bypass fixed so that I can use
it again !

By the way two suggestions to really improve AG use :
-  in the widget monitor it would be very useful to have a summary CPU load
by server, and in the detailed list too with all AG insert !
- CPU load by server is usefull but most of the drops I got in my projects
where due a single core usage spike (often close too 100% in average). It
was not due to AG. So it would be really helpfull to have that information
too, especially when you create plugin chain in AG or affect a new AG
instance to a server, because CPU usage on all cores can give some wrong
indications, and then you have drop (which again is not related to AG but
to affecting one ore more plugin to a server with some almost
saturated core)

Thanks again for your amazing work guys



Le dim. 5 sept. 2021 à 00:00, Serge Albou ***@***.***> a
écrit :

> Great for the bypass fix !
>
> I didnt had time today to do the stuff for the log, i will do my best for
> tommorow i promise
>

**Reply by Pete3030** [reply]
I would second that request regarding the CPU meter..that would be indeed very useful. Can't wait for the new Beta ;)

**Comment by Henri75** [comment]
Bad news, I still got problem with ASIO guard not functionning. The problem seems less exagerated than before, but it is still here..
On a project with half ASIO load in the meter, when I activate ASIO guard in whatever mode, the load meter doubles ! And I cant play it anymore ...
ASIO Guard should decrease the load, not increasing it. More over the meter is not stable oscillating between 75 and 100%, when it was stable at less than 50% without ASIO. Never encountered that before with any plugin.
As soon as I remove the AG inserts, everything is fine....
